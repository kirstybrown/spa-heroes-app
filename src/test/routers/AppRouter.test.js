import { mount } from 'enzyme';

import { AuthContext } from '../../auth/authContext';
import { AppRouter } from '../../routers/AppRouter';

describe('Tests in <AppRouter />', () => { 

    const contextValue = {
        user: {
            logged: false
        }
    }

    test('should show login screen if user is not authenticated', () => { 

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('h1').text().trim() ).toBe( 'Login Screen' );
        
     });



 })