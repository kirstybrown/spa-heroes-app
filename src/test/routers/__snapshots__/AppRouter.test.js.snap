// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AppRouter /> should show login screen if user is not authenticated 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
